### PR TITLE
Setting up associations

### PR DESCRIPTION
**Pull request by ChrispaixK**
# OOP school library App 📚 
> set up associations

In this PR I :
* Create `Classroom` class
* Setup relationship between `Classroom` and `Student` classes
* Create `Book` class
* Create `Rental` class 
* Setup relationship between `Person` and `Book` classes using `Rental` class as intermediate
* Fix linters errors

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+  def initialize(label, author)
+    @title = label
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(book, date)
+    Rental.new(date, self, book)
+  end
```
Comment (DammyShittu) [inline book.rb]: - [x] We already have the book object, so we don't need to provide it. Instead what we need is the person object and the date.

- [x] Also, be sure to organize your parameters based on how you defined the constructor in the rental class.

  ```
  def add_rental(person, date)
    Rental.new(date, self, person)
  end
  ```

File: rental.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+require './person'
+require './book'
+
+class Rental
+  attr_accessor :date, :book, :person
+
+  def initialize(date, book, person)
+    @date = date
+    @book = book
+    book.rental << self
```
Comment (DammyShittu) [inline rental.rb]: - [x] This should be `book.rentals`, not `book.rental` based on what you defined in the book class

File: person.rb [hunk 3]
```diff
@@ -26,4 +28,8 @@ def correct_name
   def isof_age
     @age >= 18
   end
+
+  def add_rental(person, date)
+    Rental.new(date, self, person)
+  end
```
Comment (DammyShittu) [inline person.rb]: - [x] We need the book object. not person and we need to also arrange the parameters based on the rental class constructor.

  ```
  def add_rental(book, date)
    Rental.new(date, book, self)
  end
  ```

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @ChrispaixK,

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: book.rb (hunk 1 above)
Comment (ChrispaixK) [reply book.rb]: Thanks for your help @DammyShittu ! 👍🏾

**Review by cynthiainga — APPROVED** [review]
## STATUS APPROVED ✔️ ✔️ 

Hi @ChrispaixK, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- [x] No linters errors ✔️ 
- [ ] Descriptive PR ✔️ 
- [ ] Correct Gitflow ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ChrispaixK** [comment]
Thank you @cynthiainga 😄

**Pull request merged**